src/gpu/intel/ocl/micro_sdpa.cl

```diff
@@ -392,6 +392,21 @@ micro_sdpa(const global KEY_DATA_T *K, const global half *Q,
         tile_hbroadcast_min(&S_tile, k_mask);
 #endif
 
+#if WITH_CAUSAL_MASK
+        /* Apply causal mask */
+        for (int jj = 0; jj < (ugemm_kq_c_type_block1 * ugemm_kq_c_type_nblock1); jj++) {
+          for (int ii0 = 0; ii0 < (ugemm_kq_c_type_block0 * ugemm_kq_c_type_nblock0); ii0 += SUBGROUP_SIZE) {
+            int iii = ii0 + get_sub_group_local_id();
+            int k_idx = wg_j0 + sg_j0_kq + iii;
+            int q_idx = k0 + sg_i0_kq + jj;
+
+            if(k_idx < q_idx || k_idx >= k || q_idx >= q) {
+              //printf("setting to -INFINITY k_idx: %d q_idx: %d sg_i_kq: %d sg_j_kq: %d\n", k_idx, q_idx, sg_i_kq, sg_j_kq);
+              tile_access(S_tile, ii0, jj, SUBGROUP_SIZE, ugemm_kq_c_type_block0, ugemm_kq_c_type_block1, ugemm_kq_c_type_nblock0) = -INFINITY;
```

Comment:
As I read it from the name, it seems to write inside the tiles (or squares), but tiles that have diagonal in them require a point-by-point update after the gemm was computed over the block, at least, I would expect it happening...
I'm not sure if it's done here or not, but in case it does, would be nice to have some documentation in here...

Reply:
I have updated the code to be more inline with the other tile operations. This is happening in this code but it doesn't look like it because of the way subgroups are organized.